### PR TITLE
[Merged by Bors] - Expose mint feature in bevy_math/glam

### PR DESCRIPTION
**Pull request by harudagondi**
# Objective

- Expose `mint` feature of `glam` in `bevy_math`.
- Unblocks harudagondi/bevy_oddio#22
	- [`oddio::SpatialOptions`] uses mint types

[`oddio::SpatialOptions`]: https://docs.rs/oddio/latest/oddio/struct.SpatialOptions.html

## Solution

- Added features in `bevy_math`, ~`bevy_internal`, `bevy`~
- ~Updated `docs/cargo_features.md`~

---

### Added

- `mint` feature in `bevy_math` to allow interoperation of glam types with mint-compatible libraries.

### CODE REVIEW

**Comment by harudagondi** [comment]
This is different from #5270 as this is an opt-in feature.

**Review by afonsolage — APPROVED** [review]

**Comment by mockersf** [comment]
It would make more sense for `bevy_oddio` to enable the feature on their side.

**Comment by DJMcNab** [comment]
> It would make more sense to me for bevy_oddio to enable the feature on their side.

I don't think that's right, since making users manually keep their glam version up to date seems not optimal.

Why not just expose all of the glam features in `bevy_math`

I don't think we should filter this feature all the way up to the root - it should be `bevy_math` exclusive.

**Comment by harudagondi** [comment]
> It would make more sense to me for `bevy_oddio` to enable the feature on their side.

I'm using `Vec3` internally. Also, trying to enable something like "bevy_math/glam/mint", I get this error:

```
error: failed to parse manifest at `C:\<path to>\bevy_oddio\Cargo.toml`

Caused by:
  feature `bevy_math/glam/mint` in dependency `bevy` is not allowed to contain slashes
  If you want to enable features of a transitive dependency, the direct dependency needs to re-export those features from the `[features]` table.
```

**Comment by mockersf** [comment]
What DJMcNab is suggesting is exposing a `mint` feature (and other `glam` features) on `bevy_math` but not bubbling it up to `bevy`, which I would prefer 👍 

I worry about the multiplication of features exposed by Bevy, this suggestion would fix it

**Comment by harudagondi** [comment]
> but not bubbling it up to `bevy`

Sorry if this is wrong, as I am not really familiar with how rust features work. If my library (e.g. bevy_oddio) directly uses `bevy` as a dependency, how would I enable `mint` for `bevy_math` as a feature? Do I need to reimport `bevy_math` (with the `mint` feature) along with `bevy`?

**Comment by mockersf** [comment]
> Sorry if this is wrong, as I am not really familiar with how rust features work. 

It is not wrong, it's just that features in Bevy requires some work/maintenance due to how the crates are organised. As you saw in that PR you need to modify three crates to expose a feature.
In this case, I feel that mint compatibility is not so much a feature of Bevy as a feature of the math layer, that's why it feel not good to me to expose it on Bevy directly. Others may feel differently and it would be OK to them.

> If my library (e.g. bevy_oddio) directly uses bevy as a dependency, how would I enable mint for bevy_math as a feature? Do I need to reimport bevy_math (with the mint feature) along with bevy?

Yup, you would have as dependency:
```toml
bevy = { version = "0.8", default-features = false, features = ["..."] }
bevy_math = { version = "0.8", features = ["mint"] }
```

**Review by alice-i-cecile — APPROVED** [review]
This strategy looks good to me now.

**Review by mockersf — APPROVED** [review]

File: crates/bevy_math/Cargo.toml
Comment (mockersf) [inline crates/bevy_math/Cargo.toml:16]:
```suggestion
mint = ["glam/mint"]

```

Comment (mockersf) [reply crates/bevy_math/Cargo.toml:16]: missing a new line at the end of the file

**Comment by alice-i-cecile** [comment]
@harudagondi I'll merge this once a) merge conflicts are resolved and b) you add the new line :)

**Pull request closed without merging**
